File: QytetetJava/src/modeloqytetet/TituloPropiedad.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package modeloqytetet;

/**
 *
 * @author jose26398
 */
public class TituloPropiedad {
    private String nombre;
    private boolean hipotecada;
    private int alquilerBase;
    private float factorRevalorizacion;
    private int hipotecaBase;
    private int precioEdificar;
    private Casilla casilla;
    private Jugador propietario;

    public TituloPropiedad(String nombre, int alquilerBase, float factorRevalorizacion, int hipotecaBase, int precioEdificar) {
        this.nombre = nombre;
        propietario = null;
        this.hipotecada =  false;
        this.alquilerBase = alquilerBase;
        this.factorRevalorizacion = factorRevalorizacion;
        this.hipotecaBase = hipotecaBase;
        this.precioEdificar = precioEdificar;
    }

    void cobrarAlquiler(int coste){
        propietario.modificarSaldo(coste);
    }
    
    public String getNombre() {
        return nombre;
    }

    public Casilla getCasilla() {
        return casilla;
    } 

    boolean getHipotecada() {
        return hipotecada;
    }

    int getAlquilerBase() {
        return alquilerBase;
    }

    float getFactorRevalorizacion() {
        return factorRevalorizacion;
    }

    int getHipotecaBase() {
        return hipotecaBase;
    }

    int getPrecioEdificar() {
        return precioEdificar;
    }
    
    boolean propietarioEcarcelado(){
        return propietario.getEncarcelado();
    }

    void setHipotecada(boolean hipotecada) {
        this.hipotecada = hipotecada;
    }
    
    void setCasilla(Casilla casilla){
        this.casilla = casilla;
    }
    
    void setPropietario(Jugador propietario){
        this.propietario = propietario;
    }
    
    boolean tengoPropietario(){
        return propietario != null;
    }

    @Override
    public String toString() {
        if (propietario == null)
            return "TituloPropiedad{" + "nombre=" + nombre + ", hipotecada=" + hipotecada + ", alquilerBase=" + alquilerBase + ", factorRevalorizacion=" + factorRevalorizacion + ", hipotecaBase=" + hipotecaBase + ", precioEdificar=" + precioEdificar + ", propietario=" + "NADIE" + '}';
        else
            return "TituloPropiedad{" + "nombre=" + nombre + ", hipotecada=" + hipotecada + ", alquilerBase=" + alquilerBase + ", factorRevalorizacion=" + factorRevalorizacion + ", hipotecaBase=" + hipotecaBase + ", precioEdificar=" + precioEdificar + ", propietario=" + propietario.getNombre() + '}';
    }
    
   
       
    
    
    
}

File: QytetetRuby/lib/titulo_propiedad.rb
#encoding: utf-8
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.

module ModeloQytetet
  
  class TituloPropiedad
    
    def initialize(nom, alq_base, fact_rev, hip_base, prec_ed)
      @nombre = nom
      @hipotecada = false
      @alquilerBase = alq_base
      @factorRevalorizacion = fact_rev
      @hipotecaBase = hip_base
      @precioEdificar = prec_ed     
      @casilla
      @propietario
    end
    
    
    public
    attr_reader :nombre, :alquilerBase, :factorRevalorizacion,
                        :hipotecaBase, :precioEdificar
              
    attr_accessor :hipotecada, :casilla, :propietario
    
    
    def cobrar_alquiler(coste)
      @propietario.modificar_saldo(coste)
    end
    
    
    def propietario_encarcelado()
      return @propietario.encarcelado
    end
    
    
    def tengo_propietario()
      @propietario != nil
    end
    
    
    def to_s()
      "TituloPropiedad{ nombre: #{@nombre}, hipotecada: #{@hipotecada},  alquilerBase: #{@alquilerBase},
      factorRevalorizacion: #{@factorRevalorizacion}, hipotecaBase: #{@hipotecaBase}, precioEdificar: #{@precioEdificar} }\n"
    end
    
  end

end